### ISSUE TITLE
Environment creation problem and error in the example.ipynb

### ISSUE THREAD

**Issue by panagiotoum** [IrinaStatsLab/GlucoBench#2]
1)	I attempted to create an environment using the recommended command conda env create -f environment.yml as outlined in your repository's README file (tried in both Win and Linux). However, I encountered the following errors:
![image](https://github.com/IrinaStatsLab/GlucoBench/assets/81033284/96cd19b5-34b5-473e-8848-6269d1d549bf)
 
Note that the code line that is recommended in your readme file it exports the current Conda environment's configuration to a YAML file named environment.yml, and not install the environment dependencies. Also, darts library is missing from the dependencies.

2)	Upon creating my own environment and attempting to execute example.ipynb, I encountered dimensionality issues while running cell 23. Specifically, the error occurred at the following line:
model.fit_from_dataset(dataset_train, verbose=False)
See details:
-> [18](vscode-notebook-cell:?execution_count=31&line=18) model.fit_from_dataset(dataset_train, verbose=False)

ValueError                                Traceback (most recent call last)
Cell In[31], [line 18](vscode-notebook-cell:?execution_count=31&line=18)
      [3](vscode-notebook-cell:?execution_count=31&line=3) model = NHiTSModel(input_chunk_length=formatter.params['nhits']['in_len'], 
      [4](vscode-notebook-cell:?execution_count=31&line=4)                    output_chunk_length=formatter.params['length_pred'],
      [5](vscode-notebook-cell:?execution_count=31&line=5)                    num_stacks=3, 
   (...)
     [15](vscode-notebook-cell:?execution_count=31&line=15)                    batch_size=formatter.params['nhits']['batch_size'],
     [16](vscode-notebook-cell:?execution_count=31&line=16)                    pl_trainer_kwargs={"accelerator": "gpu", "devices": [0]})
     [17](vscode-notebook-cell:?execution_count=31&line=17) # fit model
---> [18](vscode-notebook-cell:?execution_count=31&line=18) model.fit_from_dataset(dataset_train, verbose=False)

File [c:\Users\Maria](file:///C:/Users/Maria)\.conda\envs\llmtime\lib\site-packages\darts\utils\torch.py:112, in random_method.<locals>.decorator(self, *args, **kwargs)
    [110](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/utils/torch.py:110) with fork_rng():
    [111](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/utils/torch.py:111)     manual_seed(self._random_instance.randint(0, high=MAX_TORCH_SEED_VALUE))
--> [112](file:///C:/Users/Maria%20Panagiotou/.conda/envs/llmtime/lib/site-packages/darts/utils/torch.py:112)     return decorated(self, *args, **kwargs)

File [c:\Users\Maria](file:///C:/Users/Maria) \.conda\envs\llmtime\lib\site-packages\darts\models\forecasting\torch_forecasting_model.py:929, in TorchForecastingModel.fit_from_dataset(self, train_dataset, val_dataset, trainer, verbose, epochs, num_loader_workers)
    [877](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/models/forecasting/torch_forecasting_model.py:877) @random_method
    [878](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/models/forecasting/torch_forecasting_model.py:878) def fit_from_dataset(
    [879](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/models/forecasting/torch_forecasting_model.py:879)     self,
   (...)
    [885](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/models/forecasting/torch_forecasting_model.py:885)     num_loader_workers: int = 0,
    [886](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/models/forecasting/torch_forecasting_model.py:886) ) -> "TorchForecastingModel":
    [887](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/models/forecasting/torch_forecasting_model.py:887)     """
...
    [573](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/models/forecasting/pl_forecasting_module.py:573)         else past_target,
    [574](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/models/forecasting/pl_forecasting_module.py:574)         static_covariates,
    [575](file:///C:/Users/Maria/.conda/envs/llmtime/lib/site-packages/darts/models/forecasting/pl_forecasting_module.py:575)     )

ValueError: not enough values to unpack (expected 3, got 2)

This issue also persists when attempting to utilize transformers and tft.

Questions:
1)Is it possible you could provide the packages with an alternative method e.g., requirements.txt?
2) Do you have any ideas about the problem I encounter and how to resolve it?

**Comment by mrsergazinov** [comment]
Hi @panagiotoum !

First, thank you for finding our repo useful! Our hope has always been to promote openness and attract more attention to the important area of CGM understanding. 

Regarding your issues, I have pushed two updates which hopefully should fix the bugs that you have found (thanks for contributing!). Here are the details for future reference:

1. We changed the installation instructions, removing the unnecessary dependencies. The dependencies can now be installed via simple `pip install -r requirements.txt`.
2. This bug has been introduced due to the recent updates of the `darts`package. An additional end time token must now be appended to each data sample in the `__getitem__` call. 

In general, we had to re-write Darts `Dataset` classes because:

1. As of last year, Darts did not support random sampling of input-output sequences. 
2. Our goal was to be able to support both pure PyTorch and Darts models. However, the `Dataset` class for the former (PyTorch) models cannot yield anything besides `Numpy` arrays in the `__getitem__`. Darts, on the other hand, can easily handle such cases. For example, when no static features are supplied, default Darts `Dataset` gives a sample `(..., None, ...)` where `None` indicates that no static features are given. However, if we try to pass this to the default `collate` function of PyTorch, it would throw an error since `None` type is not supported.
